### DISCUSSION TITLE
AqualinkD Pi-HAT

### DISCUSSION THREAD

**Discussion by chrisdaun** [aqualinkd/AqualinkD#371] [General]
Built a Raspberry Pi HAT
Has a Buck converter circuit to power the Pi off the Aqualink bus power and an rs485 interface for communication on the bus.

In my setup I'm using an OrangePi Zero 2w on the UART5 GPIO pins but should work with a regular raspberry pi using the same pins.

This is a prototype but if there's interest I'll clean up my design files and publish them.

![image](https://github.com/user-attachments/assets/635c2cf3-2b60-47a3-82d4-3db560b4f45c)
![image](https://github.com/user-attachments/assets/ed6c2842-9963-4798-bfc6-d2efc64fc135)
![image](https://github.com/user-attachments/assets/326ea8ba-66eb-4a54-9c34-2a9c0b284250)
![image](https://github.com/user-attachments/assets/f9788930-fe72-42dd-b43c-966492eb283e)

**Comment by sfeakes** [comment]
That's totally awesome.  I've been looking for something like this for years.  

EDIT.  I'm by no means an electronics expert, but looking at the pins, this should work on any RaspberryPI.  I'd love to see the design files and who you used / how you get the board printed.

**Comment by chrisdaun** [comment]
Cool, glad there's interest 😁

I had the bare boards fabbed by jlcpcb.com. Cost was under $5 shipped for 5 PCBs (minimum order quantity).
![image](https://github.com/user-attachments/assets/0ff5c7b8-e1ef-43a2-b40d-73a56c3e8b14)

For components, I pulled them from a couple sub $2 modules I was already using in my existing AqualinkD setup (shown below).
![image](https://github.com/user-attachments/assets/af82bef2-8082-4155-a9bd-1b51284d1fd1)

**Reply by sfeakes** [reply]
Dam that's a lot of desoldering.

Looks like you used these 2.
https://www.amazon.com/KNACRO-RS485-Module-Serial-UART/dp/B078N19X3L
https://www.amazon.com/RPTCOTU-DC5V-Converter-5A-Transformer/dp/B0CSYY9138

**Reply by chrisdaun** [reply]
Yep those are them.
Much cheaper from AliExpress but those are the ones. Desoldering wasn't too bad with a hot plate.

If you didn't want to go the desoldering route you could obviously just source all the parts from the BOM but for prototyping I chose to use what I had vs paying for parts in low volume.

Using the modules as is worked great too but obviously the HAT is a cleaner solution.

I have enough parts to build another for testing if you'd like to try it out and give feedback.

This was the hot plate I used for desoldering/soldering the parts
![image](https://github.com/user-attachments/assets/086217a9-c0ce-4557-9345-60e5e56e1ec5)

**Reply by sfeakes** [reply]
I'd love to try this out. I'll shoot you an email.

**Comment by 64Spaces** [comment]
This looks awesome. Are you able or wanting to sell one? If yes, I would be excited to get one. I’m happy to purchase a board and either pay for shipping or email you a UPS shipping label.

**Reply by chrisdaun** [reply]
Hey thanks for the interest but I'm not really setup to produce/sell these currently. My goal right now is to do a little more testing with more than just my single home setup before I call it good.

I will open source the design and fabrication files for anyone that wants to build their own. Maybe if there's enough interest, myself or someone else would do a group buy kinda thing and get a bunch produced by jlcpcb.com or pcbway.com.

**Reply by 64Spaces** [reply]
Thank you. I completely understand. I look forward to seeing your progress.

**Reply by chrisdaun** [reply]
Thanks for understanding :)
Here's the repo if you want to follow along:
https://github.com/chrisdaun/pi-zero-serial-hat/tree/power_simplified

**Reply by 64Spaces** [reply]
Perfect, I just followed the project on GitHub.

**Reply by chrisdaun** [reply]
> Perfect, I just followed the project on GitHub.

These are ready now :)
https://github.com/aqualinkd/aqualinkd-zero-hat
